e2e/evm/test.go

```diff
@@ -105,3 +122,29 @@ func (s *IntegrationTestSuite) TestErc20Deposit() {
 	//Balance has increased
 	s.Equal(1, destBalanceAfter.Cmp(destBalanceBefore))
 }
+
+func (s *IntegrationTestSuite) TestGenericDeposit() {
+	hash, _ := substrateTypes.GetHash(substrateTypes.NewI64(int64(1)))
+	data := calls.ConstructGenericDepositData(hash[:])
+	err := calls.Deposit(
+		s.client,
+		s.fabric1,
+		s.gasPricer,
+		s.bridgeAddr,
+		s.genericRID,
+		2,
```

Comment:
`Deposit` also looks like it accepts a `uint8` for `destDomainID`; same as above, shouldn't this need to be wrapped in `uint8` in order to avoid type error or does the compiler just implicitly understand this?

Reply:
Compiler understands all of the `uint8` stuff implicitly